poseidon/poseidon_wrapper.go

```diff
@@ -0,0 +1,60 @@
+package poseidon
+
+import (
+	"bytes"
+	"errors"
+	"hash"
+)
+
+type digest struct {
```

Comment:
It's not a `digest`. Need a better name, maybe `hasher`

Reply:
done